From: GRUB Maintainers <pkg-grub-devel@alioth-lists.debian.net>
Date: Mon, 19 Jun 2023 17:23:24 +0200
Subject: fs-tester-time-fail

Explicitly unset SOURCE_DATE_EPOCH before running fs tests. In some
filesystem utils like mksquashfs, it will silently change behaviour
and cause timestamps to unexpectedly change. Reproducible builds are
good and useful for shipped artifacts, but this causes build-time
tests to fail.

Author: Steve McIntyre

Patch-Name: fs-tester-time-fail.patch
---
 tests/util/grub-fs-tester.in | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/tests/util/grub-fs-tester.in b/tests/util/grub-fs-tester.in
index ea8b2d1..df85a66 100644
--- a/tests/util/grub-fs-tester.in
+++ b/tests/util/grub-fs-tester.in
@@ -11,6 +11,10 @@ unset SOURCE_DATE_EPOCH
 fs="$1"
 
 builddir="@builddir@"
+
+# We can't have this set, or filesystem tests will fail
+unset SOURCE_DATE_EPOCH
+
 GRUBFSTEST="@builddir@/grub-fstest"
 GRUBPROBE="@builddir@/grub-probe"
 
